include/vulkan/utility/vk_format_utils.h

```diff
@@ -0,0 +1,2297 @@
+    FORMAT_UTILS_COMPATIBILITY_CLASS_64BIT_B12G12R12G12,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_64BIT_B16G16R16G16,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_64BIT_G10B10G10R10,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_64BIT_G12B12G12R12,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_64BIT_G16B16G16R16,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_64BIT_R10G10B10A10,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_64BIT_R12G12B12A12,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_8BIT,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_8BIT_2PLANE_420,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_8BIT_2PLANE_422,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_8BIT_2PLANE_444,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_8BIT_3PLANE_420,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_8BIT_3PLANE_422,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_8BIT_3PLANE_444,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_8BIT_ALPHA,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_96BIT,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_10X10,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_10X5,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_10X6,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_10X8,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_12X10,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_12X12,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_4X4,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_5X4,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_5X5,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_6X5,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_6X6,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_8X5,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_8X6,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ASTC_8X8,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_BC1_RGB,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_BC1_RGBA,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_BC2,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_BC3,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_BC4,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_BC5,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_BC6H,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_BC7,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_D16,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_D16S8,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_D24,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_D24S8,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_D32,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_D32S8,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_EAC_R,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_EAC_RG,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ETC2_EAC_RGBA,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ETC2_RGB,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_ETC2_RGBA,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_PVRTC1_2BPP,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_PVRTC1_4BPP,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_PVRTC2_2BPP,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_PVRTC2_4BPP,
+    FORMAT_UTILS_COMPATIBILITY_CLASS_S8
+};
+// Return the plane index of a given VkImageAspectFlagBits.
+//     VK_IMAGE_ASPECT_PLANE_0_BIT -> 0
+//     VK_IMAGE_ASPECT_PLANE_1_BIT -> 1
+//     VK_IMAGE_ASPECT_PLANE_2_BIT -> 2
+//     <any other value> -> FORMAT_UTILS_MAX_PLANES
+inline uint32_t GetPlaneIndex(VkImageAspectFlagBits aspect);
```

Comment:
One option is to return uint_max here (and to rename constant to something with INVALID or similar). Invalid program has more chances not to crash with value 3 (which is close to correct values) than with a large value. Another consideration point is to which degree should we care about backward compatibility. Using large invalid constant will help the library to be less affected by some future PLANE_3_BIT (not sure how probably is it).

Reply:
Added FORMAT_UTILS_INVALID_INDEX = 0xFFFFFFFF to make it clearer. The MAX_COMPONENTS is required since that creates the size of the components array.